### ISSUE TITLE
Windows ACME challenge unreadable : does not use inherited security

### ISSUE THREAD

**Issue by Tarjei-stavanger** [certbot/certbot#8829]
The challenge can't be read. Possibly because of the security set on the file. For some reason it has been created to not use inherited security.

Theoretically, it should be readable, but it is not.

"Everyone" access is set to read. It should possibly be set to read and execute also.

A file created with notepad in the same directory can be read with no problem.

## My operating system is (include version):
Windows Server 2012 R2 Standard

## I installed Certbot with (snap, OS package manager, pip, certbot-auto, etc):
Exe file dowloaded from certbot site.

## I ran this command and it produced this output:

C:\Program Files (x86)\Certbot\bin>.\certbot.exe certonly --webroot --debug-challenges
Saving debug log to C:\Certbot\log\letsencrypt.log
Plugins selected: Authenticator webroot, Installer None
Please enter in your domain name(s) (comma and/or space separated)  (Enter 'c' to cancel): www.stavangerbilder.no
Requesting a certificate for www.stavangerbilder.no
Performing the following challenges:
http-01 challenge for www.stavangerbilder.no
Input the webroot for www.stavangerbilder.no: (Enter 'c' to cancel): C:\Sites\Stavangerbilder
Waiting for verification...

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Challenges loaded. Press continue to submit to CA. Pass "-v" for more info about

challenges.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Press Enter to Continue
←[31mChallenge failed for domain www.stavangerbilder.no←[0m
http-01 challenge for www.stavangerbilder.no
Cleaning up challenges
←[31mSome challenges have failed.←[0m
←[1m
IMPORTANT NOTES:
←[0m - The following errors were reported by the server:

   Domain: www.stavangerbilder.no
   Type:   unauthorized
   Detail: Invalid response from
   http://www.stavangerbilder.no/.well-known/acme-challenge/9d6gUjuXg-ICSNp2p1g8_1oDdKRQXVN0KS83D1JadfM
   [91.243.69.5]: "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0
   Strict//EN\"
   \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">\r\n<html   xmlns=\"http"

   To fix these errors, please make sure that your domain name was
   entered correctly and the DNS A/AAAA record(s) for that domain
   contain(s) the right IP address.

## Certbot's behavior differed from what I expected because:
When I try to use the browser to view the challenge I get a 404 error.

http://www.stavangerbilder.no/.well-known/acme-challenge/9d6gUjuXg-ICSNp2p1g8_1oDdKRQXVN0KS83D1JadfM

A file located in the same directory retrieves fine. Possibly because the security on it is inherited.

## Here is a Certbot log showing the issue (if available):

Replay-Nonce: 0104nKk5JAx96rwBBHkqTWJZ-DcBh55A9QHlAd_6Jf-3yEw
X-Frame-Options: DENY
Strict-Transport-Security: max-age=604800

{
  "identifier": {
    "type": "dns",
    "value": "www.stavangerbilder.no"
  },
  "status": "invalid",
  "expires": "2021-05-13T05:51:38Z",
  "challenges": [
    {
      "type": "http-01",
      "status": "invalid",
      "error": {
        "type": "urn:ietf:params:acme:error:unauthorized",
        "detail": "Invalid response from http://www.stavangerbilder.no/.well-known/acme-challenge/9d6gUjuXg-ICSNp2p1g8_1oDdKRQXVN0KS83D1JadfM [91.243.69.5]: \"\u003c!DOCTYPE html PUBLIC \\\"-//W3C//DTD XHTML 1.0 Strict//EN\\\" \\\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\\"\u003e\\r\\n\u003chtml xmlns=\\\"http\"",
        "status": 403
      },
      "url": "https://acme-v02.api.letsencrypt.org/acme/chall-v3/12891198179/6SbYPw",
      "token": "9d6gUjuXg-ICSNp2p1g8_1oDdKRQXVN0KS83D1JadfM",
      "validationRecord": [
        {
          "url": "http://www.stavangerbilder.no/.well-known/acme-challenge/9d6gUjuXg-ICSNp2p1g8_1oDdKRQXVN0KS83D1JadfM",
          "hostname": "www.stavangerbilder.no",
          "port": "80",
          "addressesResolved": [
            "91.243.69.5"
          ],
          "addressUsed": "91.243.69.5"
        }
      ],
      "validated": "2021-05-06T05:53:40Z"
    }
  ]
}
2021-05-06 07:53:41,130:DEBUG:acme.client:Storing nonce: 0104nKk5JAx96rwBBHkqTWJZ-DcBh55A9QHlAd_6Jf-3yEw
2021-05-06 07:53:41,130:WARNING:certbot._internal.auth_handler:Challenge failed for domain www.stavangerbilder.no
2021-05-06 07:53:41,130:INFO:certbot._internal.auth_handler:http-01 challenge for www.stavangerbilder.no
2021-05-06 07:53:41,130:DEBUG:certbot._internal.reporter:Reporting to user: The following errors were reported by the server:

Domain: www.stavangerbilder.no
Type:   unauthorized
Detail: Invalid response from http://www.stavangerbilder.no/.well-known/acme-challenge/9d6gUjuXg-ICSNp2p1g8_1oDdKRQXVN0KS83D1JadfM [91.243.69.5]: "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">\r\n<html xmlns=\"http"

**Comment by bmw** [comment]
If you run `certbot.exe certonly --webroot -d www.stavangerbilder.no --debug-challenges --dry-run`, can you tell me the permissions on:

1. C:\Sites\Stavangerbilder
2. C:\Sites\Stavangerbilder\.well-known\acme-challenge
3. The file created by Certbot in C:\Sites\Stavangerbilder\.well-known\acme-challenge
4. Your file created in C:\Sites\Stavangerbilder\.well-known\acme-challenge using notepad that is served just fine

**Comment by Tarjei-stavanger** [comment]
You are too late. I have already done this and I have received help on this issue.

Certbot does not set proper security on the .well-known directory so that it can create the acme-challenge directory and the challenge file.

After fixing that, the problem is that certbot does not create a suitable **web.config** file to go with the challenge file. Or do the simple thing : add a .txt extension to the file.

Certbot is simply not ready for Windows.

Win-ACME works dramatically better.

**Comment by Tarjei-stavanger** [comment]
I have already used --dry-run. I have deleted the logs since they no longer serve any purpose for me.

There is no point in me doing more runs since I've made the changes I need to do in order to get it to work.

If you want to improve certbot, on Windows, try **using** it in that environment **before** letting people try it out. Microsoft will let you use Windows server for free for 180 days. That is a good deal. Install it in a virtual machine.

https://www.microsoft.com/en-us/windows-server

The developers may want to improve on the error messages by letting certbot actually tell what it is trying to do when an error occurs.

As it is, there are other tools that work a lot better and don't need to be debugged to work.
